src/librustdoc/clean/auto_trait.rs

```diff
@@ -99,29 +131,126 @@ impl<'a, 'tcx, 'rcx> AutoTraitFinder<'a, 'tcx, 'rcx> {
         let tcx = self.cx.tcx;
         let generics = self.cx.tcx.generics_of(def_id);
 
+        let ty = self.cx.tcx.type_of(def_id);
+        let mut traits = Vec::new();
+        if self.cx.crate_name != Some("core".to_string()) &&
+           self.cx.access_levels.borrow().is_doc_reachable(def_id) {
+            if let ty::TyAdt(_adt, _) = ty.sty {
+                let real_name = name.clone().map(|name| Ident::from_str(&name));
+                let param_env = self.cx.tcx.param_env(def_id);
+                for &trait_def_id in self.cx.all_traits.iter() {
+                    if !self.cx.access_levels.borrow().is_doc_reachable(trait_def_id) ||
+                       self.cx.generated_synthetics
+                              .borrow_mut()
+                              .get(&(def_id, trait_def_id))
+                              .is_some() {
+                        continue
+                    }
+                    self.cx.tcx.for_each_relevant_impl(trait_def_id, ty, |impl_def_id| {
+                        self.cx.tcx.infer_ctxt().enter(|infcx| {
+                            let generics = infcx.tcx.generics_of(impl_def_id);
+                            let trait_ref = infcx.tcx.impl_trait_ref(impl_def_id).unwrap();
+
+                            if !match infcx.tcx.type_of(impl_def_id).sty {
+                                ::rustc::ty::TypeVariants::TyParam(_) => true,
+                                _ => false,
+                            } {
+                                return
+                            }
+
+                            let substs = infcx.fresh_substs_for_item(DUMMY_SP, def_id);
+                            let ty2 = ty.subst(infcx.tcx, substs);
```

Comment:
we don't need both `ty` and `ty2`, so just name it `ty` to reduce what the reader needs to keep in the mental cache

Reply:
We do need `ty` afterwards:

```
for_: ty.clean(self.cx),
```